lib/min_heap.rb

```diff
@@ -14,18 +14,23 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn) adding element to end of array is O(1) while heap-up worst case is O(logn)
+  # Space Complexity: O(1)
```

Comment:
Remember since your `heap_up` and `heap_down` methods are recursive, you have some space complexity from use of the system stack.  So space complexity is O(log n)